index.js

```diff
@@ -72,6 +72,17 @@ EventEmitter.prototype.listeners = function listeners(event, exists) {
  * @api public
  */
 EventEmitter.prototype.emit = function emit(event, a1, a2, a3, a4, a5) {
+  return !! this.collect(event, a1, a2, a3, a4, a5);
```

Comment:
There are some issues here:

1. It makes the switch case below useless as it will always call `collect` with 6 arguments.
2. It reduces the maximum number of argments that can be emitted to 5.
3. If the listener of the event return a falsy value the return value will be `false`, but it should be `true`.